File: selfstudy/golang/cities_sum.go
/*
task 271:
Cities - Sum
You need to calculate the total sum of city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func TotalCities(populations []int) int {
	sum := 0
	for _, v := range populations {
		sum += v
	}
	return sum
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, TotalCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, TotalCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, TotalCities([]int{1, 2, 3})},
		{[]int{100, 200}, TotalCities([]int{100, 200})},
		{[]int{}, TotalCities([]int{})},
	}
	
	for idx, t := range tests {
		res := TotalCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/cities_sum.js
/*
task 71:
Cities - Sum
You need to calculate the total sum of city populations in thousands
*/

function total_cities(populations) {
  return populations.reduce((a, b) => a + b, 0);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: total_cities([10, 20, 30]) },
  { input: [50, 60, 70], expected: total_cities([50, 60, 70]) },
  { input: [1, 2, 3], expected: total_cities([1, 2, 3]) },
  { input: [100, 200], expected: total_cities([100, 200]) },
  { input: [], expected: total_cities([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('total_cities', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(total_cities(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, total_cities(t.input) === t.expected || JSON.stringify(total_cities(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done
